vendor/bat-native-ledger/src/bat/ledger/internal/credentials/credentials_common.cc

```diff
@@ -134,14 +134,34 @@ void CredentialsCommon::GetSignedCredsFromResponse(
     return;
   }
 
+  auto* signed_creds = parsed_response.FindListKey("signed_creds");
+  if (!signed_creds || signed_creds->GetList().empty()) {
+    BLOG(ledger_, ledger::LogLevel::LOG_ERROR) <<
+        "Failed to parse signed creds";
+    callback(ledger::Result::RETRY);
+    return;
+  }
+
+  auto* public_key = parsed_response.FindStringKey("public_key");
+  if (!public_key || public_key->empty()) {
+    BLOG(ledger_, ledger::LogLevel::LOG_ERROR) << "Public key is empty";
+    callback(ledger::Result::RETRY);
+    return;
+  }
+
+  auto* batch_proof = parsed_response.FindStringKey("batch_proof");
+  if (!batch_proof || batch_proof->empty()) {
+    BLOG(ledger_, ledger::LogLevel::LOG_ERROR) << "Batch proof is empty";
```

Comment:
nit: batch proof is empty or does not exist. Maybe a different log for each or a message along the lines of `Invalid batch proof`

Reply:
this is just if it's string or not and then if it's empty. Don't really see benefit of separate logs